src/test/objectstore/store_test.cc

```diff
@@ -7238,6 +7238,133 @@ TEST_P(StoreTestSpecificAUSize, BlobReuseOnOverwrite) {
   }
 }
 
+TEST_P(StoreTestSpecificAUSize, ZeroBlockDetectionSmall) {
```

Comment:
IMO the following testing scenarios (for both big and small writes) should be extended with 
1. read-after-zero-write checking, i.e. read from the object and make sure zeros are there
2. read-after-zero-overwrite case, i.e. write non-zero first, then overwrite (may be partially?) with zeros and check the content is as expected

Reply:
Done! Wrote this in a separate commit in case you want to compare to the original, but I will squash once approved.